### PR TITLE
Matter Sensor: Streamline do_configure function in air quality sensor subdriver

### PR DESCRIPTION
**Pull request by hcarter-775**
Check all that apply

# Type of Change

- [ ] WWST Certification Request
     - If this is your first time contributing code:
          - [ ] I have reviewed the README.md file
          - [ ] I have reviewed the CODE_OF_CONDUCT.md file
          - [ ] I have signed the CLA
     - [ ] I plan on entering a WWST Certification Request or have entered a request through the WWST Certification console at developer.smartthings.com
- [ ] Bug fix
- [ ] New feature
- [ ] Refactor

# Checklist

- [X] I have performed a self-review of my code
- [X] I have commented my code in hard-to-understand areas
- [X] I have verified my changes by testing with a device or have communicated a plan for testing
- [X] I am adding new behavior, such as adding a sub-driver, and have added and run new unit tests to cover the new behavior

# Description of Change
Refactor the air quality sensor configure logic for readability and clarity.

# Summary of Completed Tests
Unit test added and old unit tests continue to pass. Tested with the VDA air quality sensor as well.

### CODE REVIEW

File: drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua
Comment (nickolas-deboom) [inline drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:211]: Is this function needed?

Comment (hcarter-775) [reply drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:211]: definitely not

File: drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua
Comment (ctowns) [inline drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:257]: Should this just be `meas_find("co")` instead of `meas_find("co-")`?

```suggestion
    if profile_name == "aqs-co" or meas_find("co") or meas_find("no2") or meas_find("ozone") or meas_find("ch2o") or
```

Comment (hcarter-775) [reply drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:257]: No, but I can leave a comment to explain that decision. If it were just `meas_find("co")` that would catch `co2` as well as `co`, which we don't want if we want to copy the current logic,

Comment (ctowns) [reply drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:257]: Also, could you help me understand how the profile name `aqs-co` could be constructed from above? I'm not sure I understand why that profile is called out specifically in this conditional

Comment (hcarter-775) [reply drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:257]: The one case where `co` wouldn't be caught with a - following it is when it's the only addition,aka`aqs-co`, which is why that is also checked.

Comment (hcarter-775) [reply drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:257]: Yes! My second comment explained that, but I definitely think this should include a comment to describe this rare case

Comment (hcarter-775) [reply drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua]: @ctowns I have updated the explanation per our conversation, and I've edited the logic a little. I removed this line `profile_name == "aqs-co"` in place of `meas_name:match("co$")`, since the original conditional was actually a little wrong :sweat_smile: since it would've also needed to check the case swhere `profile_name` included the permutations `-humidity` as well as `-temp`.

File: drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua
Comment (ctowns) [inline drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:253]: You could optimize this to be
```
return string.match(meas_name, sub_name) ~= nil
```

Since we really only want to know if the substring is present, there's no need to return the found sub string.

File: drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua
Comment (ctowns) [inline drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:260]: I believe you could simplify this further by string matching this expression: `"co[^%d]?"`

This _should_ mean that we want to match any case where `co` appears and it is not (^) followed by a digit (%d), with the additional `?` at the end to account for cases where co is at the end of a string.

Lua doesn't support standard regular expressions, but it does have some functionality that you can read up on here: https://www.lua.org/manual/5.4/manual.html#6.4.1

I think distilling this to one search directly for "just co, not co2" would be more clear, along with a short comment explaining why. However, if you disagree I leave it to your discretion whichever is easier to understand.

File: drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua
Comment (hcarter-775) [reply drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:253]: I like it :+1:

File: drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua
Comment (hcarter-775) [reply drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:260]: I don't feel super strongly about it, and it would definitely shorten the commentary needed, so let's do it :+1: thanks for the lua resource!

File: drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua
Comment (ctowns) [inline drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:257]: I know this isn't necessarily the scope of this PR, but since we are calling this out directly in the code it would be good to add a test case as an example of this working.

**Review by ctowns — APPROVED** [review]
This looks good to me! Only comment is related to adding a unit test case just to validate behavior that is called out in the code.

Also, just to be sure I would test with an air purifier or the VDA so we have a real device test case before merging. I know this is strictly a refactor, but it's still good to just do a smoke test to be sure!

**Review by nickolas-deboom — APPROVED** [review]

File: drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua
Comment (hcarter-775) [reply drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:257]: Update: that suggested check `meas_find("co[^%d]?")` doesn't work unfortunately, since the `not` part doesn't reject something that's there, it just only returns what would be returned minus the part that is ^ marked. Thus, matching `co2-` would return `co`. I was playing with using `".*co[^%d].*"` as a match, but it fails when the string ends with co. And adding a ? after [^%d] in the second version makes -co2 pass. So basically lua matching functions are not complex enough to permit co but not co2 in all cases. Extra logic is needed.

Comment (hcarter-775) [reply drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:257]: This pair of checks does work though to catch the case we're discussing: `meas_find("co%-") or meas_find("co$")`

Comment (hcarter-775) [reply drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua:257]: I'm glad we made some unit tests- good nose on a possible bug :+1:

**Pull request merged**
